plugin/__init__.py

```diff
@@ -484,6 +463,34 @@ def addNote(self, note):
         return ankiNote.id
 
 
+    def addAudio(self, ankiNote, audioObjectOrList):
+        if audioObjectOrList is not None:
+            audioList = audioObjectOrList if isinstance(audioObjectOrList, list) else [audioObjectOrList]
+            for audio in audioList:
+                if audio is not None and len(audio['fields']) > 0:
+                    try:
+                        data = util.download(audio['url'])
+                        skipHash = audio.get('skipHash')
+                        if skipHash is None:
+                            skip = False
+                        else:
+                            m = hashlib.md5()
+                            m.update(data)
+                            skip = skipHash == m.hexdigest()
+
+                        if not skip:
+                            audioFilename = self.media().writeData(audio['filename'], data)
+                            for field in audio['fields']:
+                                if field in ankiNote:
+                                    ankiNote[field] += u'[sound:{}]'.format(audioFilename)
+
+                    except Exception as e:
+                        errorMessage = str(e).replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")
+                        for field in audio['fields']:
+                            if field in ankiNote:
+                                ankiNote[field] += errorMessage
```

Comment:
Is there value in putting the error in the note? It seems more useful to just return error to caller?

Reply:
To be honest, I didn't really pay attention to this, I just extracted the code segment into the `addAudio` method without making any changes to the inner workings.